### PR TITLE
choose producer based on the topic 

### PR DESCRIPTION
**Pull request by adyach**
# One-line summary
that's needed to distribute partitions accross different producers it helps to unload served partitions by single producer

### CODE REVIEW

**Comment by adyach** [comment]
👍

File: core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaFactory.java [hunk 1]
```diff
@@ -83,8 +79,8 @@ public Producer<byte[], byte[]> takeProducer() {
     }
 
     /**
-     * Release kafka producer that was obtained by {@link #takeProducer()} method. If producer was not obtained by
-     * {@link #takeProducer()} call - method will throw {@link NullPointerException}
+     * Release kafka producer that was obtained by {@link #takeProducer(String)} method. If producer was not obtained by
+     * {@link #takeProducer(String)} call - method will throw {@link NullPointerException}
```
Comment (a1exsh) [inline core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaFactory.java:83]: Never noticed, but the comment about NPE is wrong (in general, and as in not being up to date with the actual code).

File: core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java [hunk 2]
```diff
@@ -230,7 +230,7 @@ public void repartition(final String topic, final int partitionsNumber) throws C
             if (!Boolean.TRUE.equals(areNewPartitionsAdded)) {
                 throw new TopicConfigException(String.format("Failed to repartition topic to %s", partitionsNumber));
             }
-            final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer();
+            final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer(topic);
             kafkaFactory.terminateProducer(producer);
             kafkaFactory.releaseProducer(producer);
```
Comment (a1exsh) [inline core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java:235]: Why don't we throw this piece of code away?

Comment (adyach) [reply core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java:235]: you are welcome to open another PR for that

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**